tests/unit/executor/test_step_handler.py

```diff
@@ -95,9 +102,51 @@ def test_run_builtin_build(self, new_dir, mocker):
         Path("parts/p1/run").mkdir(parents=True)
         sh = _step_handler_for_step(Step.BUILD, cache_dir=new_dir)
         result = sh.run_builtin()
+        build_script_path = Path(new_dir / "parts/p1/run/build.sh")
+        environment_script_path = Path(new_dir / "parts/p1/run/environment.sh")
+
+        assert get_mode(environment_script_path) == 0o644
+        with open(environment_script_path, "r") as file:
+            assert file.read() == dedent(
+                f"""\
+                #!/bin/bash
+                set -euo pipefail
+                # Environment
+                ## Application environment
+                ## Part environment
+                export CRAFT_ARCH_TRIPLET="x86_64-linux-gnu"
+                export CRAFT_TARGET_ARCH="amd64"
+                export CRAFT_PARALLEL_BUILD_COUNT="1"
+                export CRAFT_PROJECT_DIR="{new_dir}"
+                export CRAFT_OVERLAY="{new_dir}/overlay/overlay"
+                export CRAFT_STAGE="{new_dir}/stage"
+                export CRAFT_PRIME="{new_dir}/prime"
+                export CRAFT_PART_NAME="p1"
+                export CRAFT_STEP_NAME="BUILD"
+                export CRAFT_PART_SRC="{new_dir}/parts/p1/src"
+                export CRAFT_PART_SRC_WORK="{new_dir}/parts/p1/src"
+                export CRAFT_PART_BUILD="{new_dir}/parts/p1/build"
+                export CRAFT_PART_BUILD_WORK="{new_dir}/parts/p1/build"
+                export CRAFT_PART_INSTALL="{new_dir}/parts/p1/install"
+                ## Plugin environment
+                ## User environment
+                """
+            )
+
+        assert get_mode(build_script_path) == 0o755
+        with open(build_script_path, "r") as file:
+            assert file.read() == dedent(
+                f"""\
+                #!/bin/bash
+                set -euo pipefail
```

Comment:
The lines are already in the main script, and the script that only set variables doesn't need an interpreter because it would be sourced anyway.